### PR TITLE
feat: add createExternalExtensionProvider

### PR DESCRIPTION
**Pull request by shanejonas**
- remove index.d.ts

### CODE REVIEW

File: src/createMetaMaskExternalExtensionProvider.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+const browser = detect();
+
+export default function createMetaMaskExternalExtensionProvider() {
+  let provider;
+  try {
+    const currentMetaMaskId = getMetaMaskId();
+    const metamaskPort = chrome.runtime.connect(
+      currentMetaMaskId,
+    ) as Runtime.Port;
```
Comment (shanejonas) [inline src/createMetaMaskExternalExtensionProvider.ts]: I have to cast this to play nicely with the `webextension-polyfill-ts` type that `extension-port-stream` uses, doesnt seem to match exactly to the type `chrome.runtime.connect` returns which is a `chrome.runtime.Port`

**Review by rekmarks — CHANGES_REQUESTED** [review]
Starting to look at this now. We should delete `index.d.ts` in a separate PR.

File: src/createMetaMaskExternalExtensionProvider.ts (hunk 1 above)
Comment (rekmarks) [reply src/createMetaMaskExternalExtensionProvider.ts]: Makes sense. Port- and stream-related types seem to be incompatible 100% of the time.

**Review by rekmarks — CHANGES_REQUESTED** [review]
Looks good! I have one request and one question:

Request: Since there are multiple of them, can we move all of the extension-related files into a new folder `src/extension-provider`?

Question: I'd like us to only use the `BaseProvider` instead of the inpage provider, to force consumers of this off the legacy API. What do you think of that?

**Comment by shanejonas** [comment]
index.d.ts removal in this PR: https://github.com/MetaMask/inpage-provider/pull/153

**Comment by shanejonas** [comment]
added `src/extension-provider/`

**Review by rekmarks — APPROVED** [review]
LGTM!

**Pull request merged**